kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -5,87 +5,73 @@
 
 define([
     'jquery',
-    'base/js/namespace',
+    'bluebird',
     'kbwidget',
-    'kbaseAuthenticatedWidget',
     'narrativeConfig',
     'uuid',
-    'kb_common/html',
-    'kb_sdk_clients/genericClient',
+    'common/html',
+    'common/runtime',
+    'util/string',
+    'kb_common/jsonRpc/dynamicServiceClient',
     'kb_service/client/workspace',
     'common/ui',
     'common/iframe/hostMessages',
     'common/events',
-
-    'jquery-dataTables',
+    'common/cellComponents/tabs/results/outputWidget',
 ], (
     $,
-    Jupyter,
+    Promise,
     KBWidget,
-    kbaseAuthenticatedWidget,
     Config,
     UUID,
     html,
-    GenericClient,
+    Runtime,
+    StringUtil,
+    DynamicServiceClient,
     Workspace,
     UI,
     HostMessages,
-    Events
+    Events,
+    OutputWidget
 ) => {
     'use strict';
     return KBWidget({
         name: 'kbaseReportView',
-        parent: kbaseAuthenticatedWidget,
         version: '1.0.0',
         options: {
             workspace_name: null,
             report_name: null,
-            report_window_line_height: 10,
             showReportText: true,
             showCreatedObjects: false,
             showFiles: true,
             showHTML: true,
-            inNarrative: true, // todo: toggles whether data links show in narrative or new page
             report_ref: null,
-            wsURL: Config.url('workspace'),
         },
         // workspace client
         ws: null,
+        reportData: null,
+        baseCssClass: 'kb-report-view',
         init: function (options) {
             this._super(options);
+            this.runtime = Runtime.make();
 
             // Create a message pane
             this.$messagePane = $('<div/>').addClass('kbwidget-message-pane kbwidget-hide-message');
             this.$elem.append(this.$messagePane);
 
-            this.$mainPanel = $('<div>').addClass('report-widget');
+            this.$mainPanel = $('<div>').addClass(this.baseCssClass);
             this.$elem.append(this.$mainPanel);
-
+            this.ws = new Workspace(Config.url('workspace'), {token: this.runtime.authToken()})
             return this;
         },
-        loggedInCallback: function (event, auth) {
```

Comment:
Skip all the loggedInCallback stuff now, and just fetch the auth token from Runtime. This, among other things, makes it SO MUCH EASIER to test this widget.